### DISCUSSION TITLE
How to setup a Sonoff Zigbee Bridge Pro Hub as a Router/Repeater

### DISCUSSION THREAD

**Discussion by georgelev06** [arendst/Tasmota#20466] [Q&A] [answered]
Hi,

I've done a lot of research into this but haven't been successful with setting up a Sonoff Zigbee Bridge Pro Hub as a Repeater/Router. I have a large Zigbee Network and intend to make a large scale network where my spare Sonoff Zigbee Bridge Pro Hubs can be used to make my network a bit stronger. 

I have 3x Sonoff Zigbee Bridge Pro Hubs running Tasmota [13.3.0.2(zbbrdgpro)]. 

![image](https://cdn1.espace-domotique.fr/5425-large_default/sonoff-zb-bridge-p-box-domotique-zigbee-30-wifi-pro.jpg)

- [x] **Read other Forum/Q&A Guides**
   - [Zigbee Bridge as Router for Zigbee Bridge Pro?](https://forum.ewelink.cc/t/zigbee-bridge-as-router-for-zigbee-bridge-pro/20636) - I am unaware of NSPanel Pro and it's usecase.
   - [How to Convert the Sonoff Zigbee Bridge into a Router / Repeater](https://digiblur.com/2022/02/20/how-to-convert-the-sonoff-zigbee-bridge-into-a-router-repeater/) - Only works for EFR32MG21
   - https://github.com/arendst/Tasmota/discussions/16957
   - https://github.com/arendst/Tasmota/discussions/14419
- [x] **Flashed Sonoff Hubs with Tasmota**
- [x] **Flashed with [Tasmota Zigbee Bridge Pro Firmware ](https://ota.tasmota.com/tasmota32/release/tasmota32-zbbrdgpro.bin)**
- [x] **Resized the FS Partition to 832 KBs using Partition Manager**
- [x] **Copied the following: [SonoffZBPro Router HEX](https://github.com/arendst/Tasmota/blob/development/tools/fw_SonoffZigbeeBridgePro_router_only_cc2652/SonoffZBPro_router_20220125.hex); [Sonoff ZBPro Flasher](https://github.com/arendst/Tasmota/blob/development/tasmota/berry/zigbee/sonoff_zb_pro_flasher.be); [CC2652 Flasher](https://github.com/arendst/Tasmota/blob/development/tasmota/berry/zigbee/cc2652_flasher.be); [IntelHex](https://github.com/arendst/Tasmota/blob/development/tasmota/berry/zigbee/intelhex.be)**
- [x] **Changed the Tasmota Template:**
   - {"NAME":"Sonoff Zigbee Pro","GPIO":[0, 0, 576, 0, 480, 0, 0, 0, 0, 1, 1, 5792, 0, 0, 0, 3552, 0, 320, 5793, 3584, 0, 640, 608, 32, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0 ],"FLAG":0,"BASE":1}
- [x] - Installed the HEX Firmware using 
    - import sonoff_zb_pro_flasher as cc
    - cc.load("SonoffZBPro_router_20220125.hex")
    - cc.check()
    - cc.flash()
    - 
<img width="464" alt="image" src="https://github.com/arendst/Tasmota/assets/38250098/92b02bc2-1271-4ed5-a0b7-f461b651d602">

Sadly I am now unable to connect it to Zigbee2MQTT and I am unable to Pair it. I've opened up this  Q&A for a simplistic and minimalist guide for myself and other :)

**Comment by sfromis** [comment, marked as answer]
After flashing a device as a router/repeater, it is only visible on the Zigbee mesh network, you cannot connect it to Zigbee2MQTT or do any pairing. Ideally, it should improve the mesh by providing additional routing opportunities. This is normal and expected. The Tasmota side also loses any connection to Zigbee operation, but you could use it for other purposes.

**Comment by crypto10pi** [comment]
Maybe this page will help something,
[https://digiblur.com/2022/02/20/how-to-convert-the-sonoff-zigbee-bridge-into-a-router-repeater/](https://digiblur.com/2022/02/20/how-to-convert-the-sonoff-zigbee-bridge-into-a-router-repeater/)  and the file silabs.js.

**Comment by PhillyGilly** [comment]
I had a ZB pro running under Tasmota for two years but decided to convert it to a Bridge after buying Home Assistant Yellow which has integral Zigbee co-ordinator. As I had more-or-less forgotten my earlier Tasmota experience, I needed to scratch my head in the later part of your method statement and suggest a couple of tweaks to the procedure description:
Installed all the downloaded files using the File Manager utility.
Run these four lines of code in the Berry console to activate the HEX Firmware
```
import sonoff_zb_pro_flasher as cc
cc.load("SonoffZBPro_router_20220125.hex")
cc.check()
cc.flash()
```
Wait 5-10 minutes for the device to restart.

Otherwise all good. Thanks

**Reply by Jens-Wymeersch** [reply]
I followed your steps and looked afterwards in the console (success). However when launching Z2M, I can't seem to find to find the router. Any suggestions ?

**Reply by PhillyGilly** [reply]
Your router needs to attach to the Zigbee network co-ordinator (see below) You may need to use add device to find it. There is quite a lot of waiting around for things to happen.
![image](https://github.com/user-attachments/assets/5ca1100a-7a64-4dca-b541-8a77021849ad)

**Reply by Jens-Wymeersch** [reply]
What do you mean ? I just repurposed a ZB-E as a router which Z2M found directly. But no luck with the ZBB-Pro.

**Comment by Floriky** [comment]
Hello, I am experiencing the same issue with Zigbee2MQTT on Home Assistant. When I start the pairing process, it does not detect the Sonoff device. The flashing process completed without any errors.

**Comment by ManuBT1** [comment]
Hi everyone,

I successfully (or at least I think successfully) flashed my Sonoff Zigbee Bridge Pro with the Router Only firmware. However, I’m now unsure how to activate pairing mode.

The hardware button on the device just powers it off, and there don’t seem to be any software buttons or commands available for pairing.

Does anyone know the correct way to put the device into pairing mode after flashing the Router Only firmware?

Thanks in advance!

**Reply by PhillyGilly** [reply]
I use ZHA and just use the "settings-devices- add device" to add new ZigBee devices. I think you can set one hub to be the lead for pairing, but I never bothered and once the devices are paired, they just mesh to the best ZigBee hub.

**Reply by sfromis** [reply]
I suppose that it was not about how to pair new end devices, but how to get the new router into pairing mode, and paired with the coordinator, first time.

**Reply by ManuBT1** [reply]
Yes, exactly — it’s about how to pair the coordinator and the Bridge/Router itself after flashing the Router Only firmware.

I’m trying to figure out how to get the router into pairing mode so that it can join the Zigbee network for the first time.

**Reply by sfromis** [reply]
This older thread about the topic is already linked in the first post here.
https://github.com/arendst/Tasmota/discussions/16957
Someone there seems to have had success.

**Comment by chr2002qw** [comment]
Is there a really working howto, to get a Zbbridge pro in router mode ? That all scattered threads are so confusion. 
I have no luck with latest tasmota.zbbridgepro firmware (15.3.0.4)
cc check fails with : 

15:42:49.390 FLH: Starting verification of HEX file
15:42:49.391 BRY: Exception> 'index_error' - string index out of range
15:42:49.392 stack traceback:
15:42:49.392 	<unknown source>: in function `parse`
15:42:49.393 	<unknown source>: in function `check`
15:42:49.394 	<unknown source>: in function `main`

The firmware file is from tasmota github from 2022 which is patched.
